eo-maven-plugin/src/main/java/org/eolang/maven/ChNarrow.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * The MIT License (MIT)
+ *
+ * Copyright (c) 2016-2022 Objectionary.com
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package org.eolang.maven;
+
+/**
+ * Short version of hash.
+ *
+ * @since 0.28.11
+ */
+final class ChNarrow implements CommitHash {
```

Comment:
@volodya-lombrozo What is wrong with having `CommitHash.narrow()` method instead of class?

Reply:
@mximp I would have to implement the same logic in all implementations of `CommitHash`. The next step is to add one more implementation `ChFake` - I don't wanna copy the same code from `ChNarrow`.

Reply:
@volodya-lombrozo `ChFake` would need to provide both full and short hashes as well. That's expected. Why do you think it's wrong? 
With current design the class has issues (see comment above).

Reply:
@mximp because in that case I have to write the next code twice in both implementations:
```java
hash.substring(0, 7);
```
And not only that one (we also have to check `hash` before `substring`)